core/src/main/java/com/expedia/www/haystack/client/idgenerators/RandomUUIDGenerator.java

```diff
@@ -22,7 +22,12 @@
 public class RandomUUIDGenerator implements IdGenerator {
 
     @Override
-    public UUID generate() {
+    public UUID generateTraceId() {
         return UUID.randomUUID();
     }
+
+    @Override
+    public UUID generateSpanId() {
+        return new UUID(0, UUID.randomUUID().getLeastSignificantBits());
```

Comment:
these only need to be unique inside a span so the chances of collision of ids are still nil.